patterns/java/src/refactoring_guru/patterns/decorator/example/decorators/CompressionDecorator.java

```diff
@@ -0,0 +1,64 @@
+package refactoring_guru.patterns.decorator.example.decorators;
+
+import java.io.*;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class CompressionDecorator extends DataSourceDecorator {
+    FileDataSource fileDataSource;
+
+    public CompressionDecorator(FileDataSource source) {
+        super(source);
+        this.fileDataSource = source;
+    }
+
+    @Override
+    public void writeData(String data) {
+        super.writeData(data);
+        try {
+            File file = new File(fileDataSource.path + fileDataSource.name);
```

Comment:
Okay, I guess I overlooked this when you asked about such code first time and I said that "code looks cool".

The CompressionDecorator should not know anything about data source specifics. In other words, it should work with DataSource interface, all other dependencies should be removed. The idea is to be able to compress any data source, would that be file, or database, or anything else. Your decorators should only work with methods described in DataSource interface – readData() and writeData().

If you have hard time finding a proper way to implement this in Java, please, think of alternative implementation and suggest it to me in trello.